### ISSUE TITLE
Domains: Removing the last MX record shows an incorrect email forwarding removal notice

### ISSUE THREAD

**Issue by arunsathiya** [Automattic/wp-calypso#50825]
### Steps to reproduce the behavior

- On a domain with no WordPress.com email forwarding configured, visit https://wordpress.com/domains/manage
- Add third-party email providers' MX records
- Remove the MX records
- When removing the last MX record, notice that a popup about removing WordPress.com email forwards appears incorrectly

#### What I expected to happen

No such popup must be shown because WordPress.com email forwards are not configured. The MX records added are of a third-party company as well.

#### What actually happened

Removing the last MX records (third-party MX records) shows an incorrect email forwarding removal notice.

### Context

Found by manual testing.

#### Browser / OS version

Not applicable. I was on Firefox 86 with Mac 11.2.2 though.

#### Is this specific to the applied theme? Which one?

Not applicable

#### Does this happen on simple or atomic sites or both?

Both simple and Atomic.

#### Is there any console output or error text?

None

#### Level of impact (Does it block purchases? Does it affect more than just one site?)

Not sure. 🤔 

#### Reproducibility (Consistent, Intermittent) Leave empty for consistent.

Consistent 

#### Screenshot / Video: If applicable, add screenshots to help explain your problem.

In the screenshots below, we can see that I am using SimpleLogin's (a third-party email alias service) MX records:

<img width="778" alt="Screenshot 2021-03-05 at 12 47 09 PM" src="https://user-images.githubusercontent.com/18581859/110244331-3e25b000-7f84-11eb-9da0-4a817074648b.png">

<img width="780" alt="Screenshot 2021-03-05 at 12 47 30 PM" src="https://user-images.githubusercontent.com/18581859/110244330-3d8d1980-7f84-11eb-8660-d75a4e320d8f.png">

<img width="927" alt="Screenshot 2021-03-05 at 12 47 35 PM" src="https://user-images.githubusercontent.com/18581859/110244327-3bc35600-7f84-11eb-8c2a-ea1bddd84e1b.png">

Before removing that MX record, the email forwarding page (https://wordpress.com/email/domain.com/forwarding/domain.com) shows that third-party MX record, which is correct. We can see that WordPress.com email forwards are not configured for that domain.

<img width="841" alt="Screenshot 2021-03-05 at 12 47 45 PM" src="https://user-images.githubusercontent.com/18581859/110244375-73ca9900-7f84-11eb-9dbe-80513e201619.png">

**Comment by klimeryk** [comment]
Yup, this is very much a bug. Looks like it goes all the way back to https://github.com/Automattic/wp-calypso/pull/6319? It's been always bugging me, but never got around to filing a report - thanks!

**Comment by jordesign** [comment]
Looks like this is resolved by https://github.com/Automattic/wp-calypso/pull/62870
